src/java/voldemort/store/UnreachableStoreException.java

```diff
@@ -36,4 +36,11 @@ public UnreachableStoreException(String s, Throwable t) {
         super(s, t);
     }
 
+    public static UnreachableStoreException wrap(String s, Throwable t) {
```

Comment:
It seems a bit dirty to me that the ```ThresholdFailureDetector``` relies on an exception's cause for some of its behavior... The cause should be purely informative, IMHO, not for controlling behavior. I'm not sure there is a guarantee provided that an exception's cause will preserved, since stacktraces have no such guarantee when the code JITted.

Regardless, assuming that relying on exception causes is actually clean, this wrap function is problematic, since it clips part of the exception. The part of the stack between the ```new UnreachableStoreException``` on line 41 and the cause should be omitted, which will make debugging very confusing...

It would probably be cleaner to keep all chained exceptions and recurse through them in order to find catastrophic ones. This would at least allow us to preserve full stacktraces for debugging purposes.